src/controllers/pokemon_controller.ts

```diff
@@ -28,6 +28,7 @@ export default class PokemonController {
   async show(request: Hapi.Request, h: Hapi.ResponseToolkit) {
     const search = request.params.poke;
     const pokemon = Pokemon.findById(parseInt(search))
```

Comment:
Why is this findById/pokecache still here?  It is unnecessary now that you're using the database as the datastore.

The flow here looks wrong.  You should be trying to find the pokemon in the database first, and finding it via the web service (by id or name) and storing it if it's not in the database.

Also your error handling is too vague.  Not every error that is raised (from the database, web service, etc) should just 404.  If the pokemon wasn't found in both our database or the web service then it should 404.  Otherwise, other errors should 500.